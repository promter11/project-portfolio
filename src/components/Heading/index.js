import React from "react";
import { string, number, node } from "prop-types";

const Heading = ({ children, className, level = 1 }) => {
  const Tag = "h" + level;

  return <Tag className={className}>{children}</Tag>;
};

Heading.propTypes = {
  children: node,
  className: string,
  level: number,
};

export default Heading;
