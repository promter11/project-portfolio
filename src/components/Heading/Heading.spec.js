import React from "react";
import { shallow } from "enzyme";

import Heading from "./index";

const props = {
  className: "k9Cds__Heading__1vt",
  level: 2
};

describe("Heading component", () => {
  it("renders correctly without props", () => {
    const wrapper = shallow(<Heading>Test</Heading>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly with props", () => {
    const wrapper = shallow(<Heading {...props}>Test</Heading>);

    expect(wrapper).toMatchSnapshot();
  });
});
