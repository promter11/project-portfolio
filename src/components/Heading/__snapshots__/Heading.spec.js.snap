// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component renders correctly with props 1`] = `
<h2
  className="k9Cds__Heading__1vt"
>
  Test
</h2>
`;

exports[`Heading component renders correctly without props 1`] = `
<h1>
  Test
</h1>
`;
